src/dmd/objc.d

```diff
@@ -162,10 +191,67 @@ extern(C++) abstract class Objc
     abstract void setObjc(ClassDeclaration cd);
     abstract void setObjc(InterfaceDeclaration);
 
+    /**
+     * Deprecate the given Objective-C interface.
+     *
+     * Representing an Objective-C class as a D interface has been deprecated.
+     * Classes have now been properly implemented and the `class` keyword should
+     * be used instead.
+     *
+     * In the future, `extern(Objective-C)` interfaces will be used to represent
+     * Objective-C protocols.
+     *
+     * Params:
+     *  interfaceDeclaration = the interface declaration to deprecate
+     */
+    abstract void deprecate(InterfaceDeclaration interfaceDeclaration) const;
+
     abstract void setSelector(FuncDeclaration, Scope* sc);
     abstract void validateSelector(FuncDeclaration fd);
     abstract void checkLinkage(FuncDeclaration fd);
 
+    /**
+     * Returns `true` if the given function declaration is virtual.
+     *
+     * Function declarations with Objective-C linkage and which are static or
+     * final are considered virtual.
+     *
+     * Params:
+     *  functionDeclaration = the function declaration to check if it's virtual
+     *
+     * Returns: `true` if the given function declaration is virtual
+     */
+    abstract bool isVirtual(const FuncDeclaration functionDeclaration) const;
+
+    /**
+     * Gets the parent of the given function declaration.
+     *
+     * Handles Objective-C static member functions, which are virtual functions
+     * of the metaclass, by returning the parent class declaration to the
+     * metaclass.
+     *
+     * Params:
+     *  functionDeclaration = the function declaration to get the parent of
+     *  classDeclaration = the current parent, i.e. the class declaration the
+     *      given function declaration belongs to
+     *
+     * Returns: the parent
+     */
+    abstract ClassDeclaration getParent(FuncDeclaration functionDeclaration,
+        ClassDeclaration classDeclaration) const;
```

Comment:
Long names imply a global name, short names imply locals. These are locals.

> cdecl is used throughout the DMD source code. 

I can't find a single instance of it. Commonly used dmd local names are:

1. ad = Aggregate Declaration
2. sd = StructDeclaration
3. cd = ClassDeclaration

This convention makes code much easier to read.

Reply:
> Long names imply a global name, short names imply locals. These are locals.

Long names imply descriptive names. I used the descriptive names in the abstract base class because those are the names that will be visible when the documentation is generated. For an outside user it's going to be much more clear. For anyone already familiar with the specific DMD naming conventions, I don't see how it can be any confusing, because `classDeclaratiaon` cannot mean anything else than "class declaration".

But fine, I'll rename it.

Reply:
Renamed.